.conf/dps_27/nginx.tasmoadmin.conf

```diff
@@ -0,0 +1,42 @@
+location ^~ /tasmoadmin {
+	# Manually deny some paths to ensure Tasmoadmin security
+	location ~* ^/tasmoadmin/(?:\.|data) {
+		deny all;
+	}
+
+	# Enable URL rewrites, required for sync clients
+	try_files $uri $uri/ /pydio/index.php?$args;
+
+	location ~ .php$ {
+		fastcgi_split_path_info ^(.+\.php)(/.*)$;
+		include fastcgi_params;
+		fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+		fastcgi_param SCRIPT_NAME $fastcgi_script_name; # necessary for owncloud to detect the contextroot https://github.com/owncloud/core/blob/v10.0.0/lib/private/AppFramework/Http/Request.php#L603
+		fastcgi_param PATH_INFO $fastcgi_path_info;
+		fastcgi_param HTTPS $https;
+		fastcgi_param modHeadersAvailable true; #Avoid sending the security headers twice
+		# EXPERIMENTAL: active the following if you need to get rid of the 'index.php' in the URLs
+		fastcgi_param front_controller_active true;
+		fastcgi_read_timeout 180; # increase default timeout e.g. for long running carddav/ caldav syncs with 1000+ entries
+		fastcgi_pass php;
+		fastcgi_intercept_errors on;
+		fastcgi_request_buffering off;
+	}
+
+	location ~ ^(.*)\.(css|js|gif||jpe?g|png|json|cache\.json)$ {
+	}
+
+	location /tasmoadmin {
+		rewrite ^/login$ /login.php last;
+		rewrite ^/logout$ /login.php?logout=logout last;
+		rewrite ^/doAjaxAll$ /index.php?doAjaxAll=doAjaxAll last;
+		rewrite ^/doAjax$ /index.php?doAjax=doAjax last;
+		rewrite "/([a-z]{2})/" /index.php?lang=$1 last;
+		rewrite ^/([a-zA-Z_]+)/([a-zA-Z_]+)/([0-9_]+)/?$ /index.php?page=$1&action=$2&device_id=$3;
+		rewrite ^/([a-zA-Z_]+)/(force)/?$ /index.php?page=$1&force=1;
+		rewrite ^/([a-zA-Z_]+)/([a-zA-Z_]+)/?$ /index.php?page=$1&action=$2;
+		rewrite ^/([a-zA-Z_]+)/([0-9]+)/?$ /index.php?page=$1&device_id=$2;
+		rewrite ^/([a-zA-Z_]+)/?$ /index.php?page=$1;
+	}
```

Comment:
@svh1985 
- As you are already inside /tasmoadmin location directive, the above second one is obsolete, as it will match in every case? Or is this to prevent even parsing the rewrite rules if one of the above locations is matched?
- The `/tasmoadmin` subdir needs to be inside the rewrite origin and target, like it is in the Lighttpd config above.
- I am not 100% sure how things are called within Nginx, but AFAIK all the rewrite rules need to be in front of the *.php location directive since it's not called anymore otherwise, at least for the rewrite rules without `last` flag? But should be simply tested. If it indeed works or is even required, then you could replace "last" by "break" which AFAIK means that no further rewrite rules are parsed, which just makes sense in those cases.
________
Btw just found that Apache `FollowSynLinks` is required for rewrites to work: https://httpd.apache.org/docs/current/mod/mod_rewrite.html#rewriterule
Didn't know that 🙂.